internal/test/framework/windowsvm.go

```diff
@@ -181,53 +181,58 @@ func (w *windowsVM) RetrieveFiles(remoteDir, localDir string) error {
 	// Get the list of all files in the directory
 	remoteFiles, err := sftp.ReadDir(remoteDir)
 	if err != nil {
-		return fmt.Errorf("error opening remote file: %v", err)
+		return fmt.Errorf("error opening remote directory %s: %v", remoteDir, err)
 	}
 
 	for _, remoteFile := range remoteFiles {
```

Comment:
@mansikulkarni96 Have you looked at the ftp client `Walk()` function? Whats the downside of using that here instead of writing your own?

Reply:
I tried using the walk function initially but it takes the argument root string and tries to find this remote root directory path that we have using lstat on local and then applies the function we define inside it. So this was increasing the complexity for a case where we first establish a sftp connection and get a []os.fileInfo object in order to read this remote path(VM).